src/components/metadata/index.js

```diff
@@ -0,0 +1,248 @@
+import PropTypes from 'prop-types'
+import React, { PureComponent } from 'react'
+import map from 'lodash/map'
+import sortBy from 'lodash/sortBy'
+import styled from 'styled-components'
+import mq from '../../utils/media-query'
+
+const _ = {
+  map,
+  sortBy,
+}
+
+const MetadataContainer = styled.aside`
+  max-width: 180px;
+  letter-spacing: 0.4px;
+  ${mq.hdOnly`
+    max-width: 250px;
+  `}
+`
+
+const SeparationLine = styled.div`
+  width: 100%;
+  height: 12px;
+  border-right: solid 0.5px #d8d8d8;
+`
+
+const UpperSeparationLine = styled(SeparationLine)`
+  border-top: solid 0.5px #d8d8d8;
+  margin-bottom: -2px;
+`
+
+const LowerSeparationLine = styled(SeparationLine)`
+  border-bottom: solid 0.5px #d8d8d8;
+`
+
+const ColoredText = styled.div`
+  color: ${props => props.color};
+  font-size: ${props => props.fontSize}px;
+  font-weight: ${props => props.fontWeight || 'normal'};
+`
+
+const CategoryFlexBox = styled.div`
+  display: flex;
+`
+
+const CategoryFlex = styled.div`
+  flex-grow: 1;
+`
+
+const CategoryText = styled(ColoredText)`
+  font-size: 16px;
+  line-height: 1.5;
+  padding-left: 5px;
+`
+
+const AuthorSection = styled.div`
+  margin-top: 15px;
+  margin-bottom: 45px;
+`
+
+const AuthorFlexBox = styled.div`
+  display: flex;
+  margin-bottom: 6px;
+`
+
+const NoShrinkFlexItem = styled.div`
+  flex-shrink: 0;
+`
+
+const authorTextHeight = 20 // px
+
+const AuthorJobTitle = styled(ColoredText)`
+  margin-left: ${props => props.marginLeft || '0'}px;
+  line-height: ${props => authorTextHeight / props.fontSize};
+`
+
+const AuthorName = styled(AuthorJobTitle)`
+  ${mq.mobileOnly`
+    display: inline-block;
+  `}
+`
+
+const AngledSeparationLine = styled.div`
+  border-bottom: 0.5px solid ${props => props.color};
+  width: 27.7px;
+  height: ${authorTextHeight / 2}px;
+  transform: rotate(-25deg);
+  margin-left: -18px;
+`
+
+const TagButton = styled.div`
+  border: solid 1px #808080;
+  border-radius: 50px;
+  padding: 5px 10px 5px 10px;
+  font-size: 14px;
+  font-weight: normal;
+  color: #808080;
+  margin-bottom: 10px;
+  margin-right: 10px;
+
+  &:before {
+    content: '#';
+  }
+`
+
+const TagsSection = styled.div`
+  display: flex;
+  flex-wrap: wrap;
+  align-content: center;
+`
+
+const StyledA = styled.a`
+  text-decoration: none;
+`
+
+const authorPropType = PropTypes.shape({
+  id: PropTypes.string.isRequired,
+  name: PropTypes.string.isRequired,
+})
+
+class Metadata extends PureComponent {
+  static propTypes = {
+    categories: PropTypes.arrayOf(
+      PropTypes.shape({
+        id: PropTypes.string.isRequired,
+        name: PropTypes.string.isRequired,
+        sort_order: PropTypes.number.isRequired,
+      })
+    ),
+    tags: PropTypes.arrayOf(
+      PropTypes.shape({
+        id: PropTypes.string.isRequired,
+        name: PropTypes.string.isRequired,
+      })
+    ),
+    themeColor: PropTypes.string.isRequired,
```

Comment:
I suggest to use the `ThemeProvider` of `styled-components` to store all theme related variables (which will be provided by the consumer of this package), then we can use these variables directly in any styled component at any nesting level and do not need to pass them via props.

Reply:
See https://github.com/twreporter/react-article-components/pull/3/files